homeassistant/components/netgear/switch.py

```diff
@@ -106,3 +195,67 @@ def async_update_device(self) -> None:
             self._state = None
         else:
             self._state = self._device[self.entity_description.key] == "Allow"
+
+
+class NetgearRouterSwitchEntity(NetgearRouterEntity, SwitchEntity):
+    """Representation of a Netgear router switch."""
+
+    _attr_entity_registry_enabled_default = False
+    entity_description: NetgearSwitchEntityDescription
+
+    def __init__(
+        self,
+        router: NetgearRouter,
+        entity_description: NetgearSwitchEntityDescription,
+    ) -> None:
+        """Initialize a Netgear device."""
+        super().__init__(router)
+        self.entity_description = entity_description
+        self._name = f"{router.device_name} {entity_description.name}"
+        self._unique_id = f"{router.serial_number}-{entity_description.key}"
+
+        self._state = None
+        self._available = False
+
+    @property
+    def is_on(self):
+        """Return true if switch is on."""
+        return self._state
+
+    @property
+    def available(self):
+        """Return if the switch is available."""
+        return self._available
+
+    async def async_added_to_hass(self):
+        """Fetch state when entity is added."""
+        await self.async_update()
+        await super().async_added_to_hass()
+
+    async def async_update(self):
+        """Poll the state of the switch."""
+        async with self._router.api_lock:
+            response = await self.hass.async_add_executor_job(
+                self.entity_description.update(self._router)
+            )
+        if response is None:
+            self._available = False
+        else:
+            self._state = response
+            self._available = True
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the switch on."""
+        async with self._router.api_lock:
+            await self.hass.async_add_executor_job(
+                self.entity_description.action(self._router), True
+            )
+        self.async_schedule_update_ha_state(force_refresh=True)
```

Comment:
Can't you assume it succeeded?
```suggestion
        self._is_on = True
```

Reply:
Actually, you don't need this. This entity has `self._attr_should_poll = True` (because that's the default) so it will already do this after the service call.
```suggestion
```

Reply:
In testing the state of the switch would flip when turned on (so it is OFF, you turn it ON, status changes to ON, few seconds later status flips to OFF and another few seconds later the status will flip to ON).

Therefore I added this async_schedule_update_ha_state to force a async_update call.

Of course I could also just assume the state to have changed, but I thought that was against HA best practices.

What do you prefer @balloob, keep it like this or assume the state has changed?

Reply:
Start with verifying that we don't already call update automatically after this service call.

Reply:
Right now you call it twice.

Reply:
@balloob I do not see `async_schedule_update_ha_state` in the ToggleEntity, Entity or the SwitchEntity classes.
I am not so familiar with the code path, could you point me at the right direction?

Reply:
You are totally right, my appologies!
I tested again and can indeed see that it polls directly after a change.
Not sure why I saw the weird behaviour before, but it is now gone.